pkg/plugin/datasource.go

```diff
@@ -225,10 +223,16 @@ func buildFrameSeries(result *axiQuery.Result) *data.Frame {
 			values = append(values, series.StartTime)
 			for _, field := range result.GroupBy {
 				v := g.Group[field]
+				// TODO: look more into it why Group[field] equal nil.
+				// ignore when a value is nil. to avoid having mismatch values and fields length which causes internal server error
+				if v == nil {
+					continue
```

Comment:
We can't just continue here, because the values slice has to match the fields slice.  Instead we could consider just appending "" for now, while we investigate specific queries that hit this

Reply:
I am really confused here, not sure what's best to be done here.
if we appended "", the values length going to be 3 and the fields length going to be 2.
which will crash in AppendRow because Fields[2] is nil
```
func (f *Frame) AppendRow(vals ...interface{}) {
	for i, v := range vals {
		f.Fields[i].vector.Append(v)
	}
}
```